### PR TITLE
fix(replays): Run TrimmingProcessor on replays

### PR DESCRIPTION
**Pull request by iker-barriocanal**
Although some fields in replays have `max_chars` defined, the trimming processor isn't run and thus the limit is not enforced. This PR runs the processor and enforces that limit.

### CODE REVIEW

File: relay-server/src/services/processor/replay.rs [hunk 1]
```diff
@@ -175,12 +175,14 @@ fn process_replay_event(
     let mut replay =
         Annotated::<Replay>::from_json_bytes(payload).map_err(ReplayError::CouldNotParse)?;
 
+    if replay.value().is_none() {
+        return Err(ReplayError::NoContent);
+    }
+
     if let Some(replay_value) = replay.value_mut() {
```
Comment (jjbayer) [inline relay-server/src/services/processor/replay.rs]: This should work, `replay_value` goes out of scope before `replay` is used again.

```suggestion
    let Some(replay_value) = replay.value_mut() else {
        return Err(ReplayError::NoContent);
    }
    replay::validate(replay_value)?;
    replay::normalize(&mut replay, client_ip, user_agent);
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+**Bug fixes**:
+
+- Trim some fields in replays to the maximum expected length. ([#3706](https://github.com/getsentry/relay/pull/3706))
```
Comment (jjbayer) [inline CHANGELOG.md]: It applies to all fields that have a `max_size`, right?

```suggestion
- Trim fields in replays to the maximum expected length. ([#3706](https://github.com/getsentry/relay/pull/3706))
```

**Review by jjbayer — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (iker-barriocanal) [reply CHANGELOG.md]: > It applies to all fields that have a max_size, right?

Yes, but not all fields have a max_size. I'm open to change it but not sure if it's better.

Comment (jjbayer) [reply CHANGELOG.md]: `some` sounds arbitrary to me, that's why I would remove it. Maybe 
```suggestion
- Trim fields in replays to their defined maximum length. ([#3706](https://github.com/getsentry/relay/pull/3706))
```

File: relay-event-normalization/src/replay.rs [hunk 3]
```diff
@@ -462,4 +469,32 @@ mod tests {
         let validation_result = validate(replay.value_mut().as_mut().unwrap());
         assert!(validation_result.is_err());
     }
+
+    #[test]
+    fn test_maxchars_trimming() {
+        let json = format!(r#"{{"dist": "{}"}}"#, "0".repeat(100));
+        let mut replay = Annotated::<Replay>::from_json(json.as_str()).unwrap();
```
Comment (bruno-garcia) [inline relay-event-normalization/src/replay.rs:476]: test only stresses dist, should it test the other fields too?

**Review by cmanallen — APPROVED** [review]
As Bruno said more test coverage is appreciated if necessary.

The max length is defined by the annotation on the fields correct?  Does this mean string trimming wasn't working before?

**Comment by iker-barriocanal** [comment]
> The max length is defined by the annotation on the fields correct?

Yep, that's correct.

> Does this mean string trimming wasn't working before?

Yes, let me know if I should clarify further the PR description.

> test only stresses dist, should it test the other fields too?

The test validates that the trimming processor runs on replays, and there are separate tests to validate that the trimming processor trims fields with `max_chars`, so it's ok to move forward with the current state. We can add more tests if there are known edge cases.
